src/main/java/de/azapps/kafkabackup/sink/BackupSinkTask.java

```diff
@@ -11,23 +12,30 @@
 import org.apache.kafka.common.TopicPartition;
 import org.apache.kafka.connect.sink.SinkRecord;
 import org.apache.kafka.connect.sink.SinkTask;
+import org.apache.kafka.connect.sink.SinkTaskContext;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
 import java.io.IOException;
 import java.nio.file.Files;
 import java.nio.file.Path;
 import java.nio.file.Paths;
-import java.util.Collection;
-import java.util.HashMap;
-import java.util.Map;
+import java.util.*;
 
 public class BackupSinkTask extends SinkTask {
     private static final Logger log = LoggerFactory.getLogger(BackupSinkTask.class);
     private Path targetDir;
     private Map<TopicPartition, PartitionWriter> partitionWriters = new HashMap<>();
     private long maxSegmentSizeBytes;
     private OffsetSink offsetSink;
+    private SinkTaskContext sinkTaskContext;
+    private BackupSinkConfig config;
+    private AdminClient adminClient;
+    private Map<TopicPartition, Long> endOffsets;
+    private Map<TopicPartition, Long> currentOffsets = new HashMap<>();
+    private EndOffsetReader endOffsetReader;
+    private String bootstrapServers;
+    private java.util.function.Consumer<Integer> exitFunction;
```

Comment:
I do not know whether this is such a good idea. We are basically killing-off Kafka Connect. For the restore procedure I implemented the termination differently: https://github.com/itadventurer/kafka-backup/blob/master/bin/restore-standalone.sh#L277

Tl;dr:
* Print a specific message when the restore is finished
* The `restore-standalone.sh` is looking for the message
* `kill` Kafka Connect and give it time to terminate it gracefully

Reply:
I understand and totally agree that this is not ideal. However, I left this in because kafka connect seems to terminate reasonably graceful, judging by the console output. I assume they installed some JVM shutdown hook. The wait-for-stdout solution in bash is definitely not ideal either. If you prefer that solution, then I will implement that, but also consider the System.exit approach.